R/appMetadata.R

```diff
@@ -67,6 +67,14 @@ appMetadata <- function(appDir,
       appDir = appDir,
       appPrimaryDoc = appPrimaryDoc
     )
+    if (appMode == "quarto-shiny") {
+      if (!"knitr" %in% quartoInfo[["engines"]]) {
+        cli::cli_abort(c(
+          "The Quarto document requires a server but does not use an executable engine.",
+          "Consider including some executable code, specifying an engine, or removing the server configuration."
+        ))
+      }
+    }
```

Comment:
@aronatkins Hmm. Its placement _does_ feel a little arbitrary, I see what you mean. And I don't see that many app-mode-specific errors here. But this _is_ where the requisite info is gathered, and just glancing over other `cli_abort` calls in the code, I don't see an overarching strategy here, so… I think it's probably fine.

Reply:
Could a `quarto-shiny` file specify an engine other than `knitr`, e.g. Python? The error talks about needing to use "an executable engine" but the code checks specifically for `knitr`.

Reply:
Good question. Yes, they may use the jupyter or knitr engines.

I've expanded the check to allow the jupyter engine, but am not making any changes to the downstream logic. rsconnect generally only supports the R engine for Shiny Quarto documents.